fiona/_shim2.pyx

```diff
@@ -138,3 +138,28 @@ cdef void set_proj_search_path(object path):
 
 cdef (int, int, int) get_proj_version():
     return (-1, -1, -1)
+
+
+cdef void set_field_datetime(void *cogr_feature, int iField, int nYear, int nMonth, int nDay, int nHour, int nMinute, int nSecond, int nMicrosecond, int nTZFlag):
+    if nMicrosecond > 0:
+        fSecond = float(nSecond + nMicrosecond / 10**6)
+    else:
+        fSecond = float(nSecond)
+    OGR_F_SetFieldDateTimeEx(cogr_feature, iField, nYear, nMonth, nDay, nHour, nMinute, fSecond, nTZFlag)
+
+
+cdef (int, int, int, int, int, int, int, int, int) get_field_as_datetime(void *cogr_feature, int iField):
+    cdef int retval
+    cdef int nYear = 0
+    cdef int nMonth = 0
+    cdef int nDay = 0
+    cdef int nHour = 0
+    cdef int nMinute = 0
+    cdef float fSecond = 0.0
+    cdef int nTZFlag = 0
+
+    retval = OGR_F_GetFieldAsDateTimeEx(cogr_feature, iField, &nYear, &nMonth, &nDay, &nHour, &nMinute, &fSecond, &nTZFlag)
+    nSecond = int(fSecond)
+    nMicrosecond = int(round((fSecond%1) * 10**6))
```

Comment:
I'm not that comfortable with the round() here. However, it was necessary, as e.g. 5.220 seconds is converted to 5.2199... internally by gdal.  (Or I do something wrong)